crates/burrego/src/evaluator.rs

```diff
@@ -1,55 +1,112 @@
 use crate::builtins;
+use crate::errors::{BurregoError, Result};
 use crate::host_callbacks::HostCallbacks;
 use crate::opa_host_functions;
 use crate::policy::Policy;
 use crate::stack_helper::StackHelper;
-use anyhow::{anyhow, Result};
+
 use itertools::Itertools;
 use std::collections::{HashMap, HashSet};
-use std::path::Path;
 use tracing::debug;
 use wasmtime::{Engine, Instance, Linker, Memory, MemoryType, Module, Store};
 
+macro_rules! set_epoch_deadline_and_call_guest {
+    ($epoch_deadline:expr, $store:expr, $code:block) => {{
+        if let Some(deadline) = $epoch_deadline {
+            $store.set_epoch_deadline(deadline);
+        }
+        $code
+    }};
+}
+
+struct EvaluatorStack {
+    store: Store<Option<StackHelper>>,
+    instance: Instance,
+    memory: Memory,
+    policy: Policy,
+}
+
 pub struct Evaluator {
-    #[allow(dead_code)]
     engine: Engine,
-    #[allow(dead_code)]
-    linker: Linker<Option<StackHelper>>,
+    module: Module,
     store: Store<Option<StackHelper>>,
     instance: Instance,
     memory: Memory,
     policy: Policy,
+    host_callbacks: HostCallbacks,
+    /// used to tune the [epoch
+    /// interruption](https://docs.rs/wasmtime/latest/wasmtime/struct.Config.html#method.epoch_interruption)
```

Comment:
I have a question about this documentation formatting. Markdown link syntax work some where when reading a source code file?

Reply:
I think it depends on the IDE being used by the developer. However it works when generating the html documentation via `cargo doc`.